core/contracts/FixedPoint.sol

```diff
@@ -97,6 +97,20 @@ library FixedPoint {
         return Unsigned(a.rawValue.mul(b));
     }
 
+    /** @dev Multiplies two `Unsigned`s, reverting on overflow, and rounds the resultant product up rather than by default, down. */
+    function mulCeil(Unsigned memory a, Unsigned memory b) internal pure returns (Unsigned memory) {
+        // To ensure rounding up occurs post-truncation, add one half of the least significant digit remaining post-truncation
+        // (i.e., the 18th digit). If the 18th digit is [0,1,2,3,4], then it will not affect the 17th digit and will get truncated
+        // down and the product will get "rounded down". If the 18th digit is [5,6,7,8,9], then the 17th digit will be increased
+        // by one before getting truncated. Therefore the resultant product will be "rounded up"
+        return Unsigned(a.rawValue.mul(b.rawValue).add(0.5 * 10**18) / FP_SCALING_FACTOR);
```

Comment:
Overall comment: I think this is a good solution.

nit: what does 0.5 * 10**18 do? How does solidity reference a decimal number...? I'd be interested to know the intermediate types here?

Reply:
so `.add()` is a SafeMath method and takes a `uint`. 

`a.rawValue.mul(b.rawValue)` produces a 36 digit number, so by adding `0.5e18` I essentially add 5 to the 19th digit. Then we truncate by dividing by `FP_SCALING_FACTOR`.

Reply:
Sorry, I think I worded that badly. Is there any documentation around what `0.5` is in solidity? As far as I understand, solidity does not support floating point numbers at all.

Reply:
yeah `0.5` and floats are not supported in general thus why I multiply by `10**18`